src/qibo/backends/__init__.py

```diff
@@ -78,39 +78,73 @@ class GlobalBackend(NumpyBackend):
         {"backend": "pytorch"},
     ]
 
-    def __new__(cls):
-        if cls._instance is not None:
-            return cls._instance
+    @classmethod
+    def backend(cls):  # pragma: no cover
+        if cls._backend is not None:
+            return cls._backend
 
         backend = os.environ.get("QIBO_BACKEND")
-        if backend:  # pragma: no cover
+        if backend:
             # Create backend specified by user
             platform = os.environ.get("QIBO_PLATFORM")
-            cls._instance = construct_backend(backend, platform=platform)
+            cls._backend = construct_backend(backend, platform=platform)
         else:
             # Create backend according to default order
             for kwargs in cls._default_order:
                 try:
-                    cls._instance = construct_backend(**kwargs)
+                    cls._backend = construct_backend(**kwargs)
                     break
-                except (ImportError, MissingBackend):
+                except (ModuleNotFoundError, ImportError):
                     pass
 
-        if cls._instance is None:  # pragma: no cover
+        if cls._backend is None:
             raise_error(RuntimeError, "No backends available.")
 
-        log.info(f"Using {cls._instance} backend on {cls._instance.device}")
-        return cls._instance
+        log.info(f"Using {cls._backend} backend on {cls._backend.device}")
+        return cls._backend
 
     @classmethod
     def set_backend(cls, backend, **kwargs):  # pragma: no cover
         if (
-            cls._instance is None
-            or cls._instance.name != backend
-            or cls._instance.platform != kwargs.get("platform")
+            cls._backend is None
+            or cls._backend.name != backend
+            or cls._backend.platform != kwargs.get("platform")
         ):
-            cls._instance = construct_backend(backend, **kwargs)
-        log.info(f"Using {cls._instance} backend on {cls._instance.device}")
+            cls._backend = construct_backend(backend, **kwargs)
+            log.info(f"Using {cls._backend} backend on {cls._backend.device}")
+        else:
+            log.info(f"Backend {backend} is already loaded.")
+
+    @classmethod
+    def get_backend(cls):
+        return cls._backend
+
+    @classmethod
+    def transpiler(cls):  # pragma: no cover
+        from qibo.transpiler.pipeline import Passes
```

Comment:
Why can't you import top-level?

Reply:
Because importing Passes at the top level causes a circular import. I tried removing it but I encountered numerous circular import issues. One example is:

```py
tests/conftest.py:10: in <module>
    from qibo.backends import construct_backend
src/qibo/__init__.py:5: in <module>
    from qibo import (
src/qibo/gates/__init__.py:1: in <module>
    from qibo.gates.channels import *
src/qibo/gates/channels.py:9: in <module>
    from qibo.backends import _check_backend
src/qibo/backends/__init__.py:14: in <module>
    from qibo.transpiler.pipeline import Passes
src/qibo/transpiler/__init__.py:1: in <module>
    from qibo.transpiler.optimizer import Preprocessing, Rearrange
src/qibo/transpiler/optimizer.py:5: in <module>
    from qibo.models import Circuit
src/qibo/models/__init__.py:1: in <module>
    from qibo.models import hep, tsp
src/qibo/models/hep.py:4: in <module>
    from qibo.backends import matrices
E   ImportError: cannot import name 'matrices' from partially initialized module 'qibo.backends' (most likely due to a circular import) (/Users/brysonkim/Desktop/intern/qibo/src/qibo/backends/__init__.py)
```

Reply:
Yes, I see.

Here there is clearly a problem in structuring the package, because we're abusing `src/qibo/backends/__init__.py`.

We should have done this even before, but since you're working on it, it could be worth to move all the content from here to a `src/qibo/backends/global.py` (or the module name you prefer, even `default.py`/`manager.py`/...), and then re-export its content in `src/qibo/backends/__init__.py`.

Unfortunately, the other point is that we should either split `matrices` from the `backends` subpackage (which is a much greater effort I'd save you, and this PR is not the best place to do that).
Indeed, even while loading `qibo.backends.global`, it will first load `qibo.backends` (that's how Python's import mechanism works with subpackages), and then trigger the same issue. Even having the `Circuit` at the same level of `hep` is another issue, but this is an issue that we'll solve in the `qibo-core` scope.

Reply:
I think it is better to update this after once _Global implementation is stable (after all reviews are clear).

Reply:
Also covered by #1490, I'm resolving.